README.md

```diff
@@ -33,70 +33,72 @@ We will define a `set` and fill it up with the tokens which current query should
 
 ```python
 # fill up global tokens in our `set`
 global_tokens = ["<something>", "....."]
 key_tokens.update(global_tokens)
 ```
 
-* Introduce some random tokens which will transfer information by transfering to other tokens which in turn can transfer to other tokens. This may reduce the cost of information travel from one token to other. This is why `random` attention attention is introduced.
+* Introduce some random tokens which will transfer information by transfering to other tokens which in turn can transfer to other tokens. This may reduce the cost of information travel from one token to other.
 
 ```python
 # Let's add random tokens to our `set`
 random_tokens = ["<something>", "....."]
 key_tokens.update(random_tokens)
 ```
 
-Now, we just need our token to attend this `set` & possibly it will represent all the tokens nicely. Similar thing we will do for all the queries & attention matrix can be completed. But remember whole point is to do it as efficiently (& fastly) as we can & this is when bigbird block sparse attention comes into picture.
+Now, we just need our token to attend this `set` & possibly it will represent all the tokens nicely. The same approach is used for all other tokens. 
+But remember, the idea here is to approximate `BERT`'s full attention as efficiently as possible. This is when `BigBird`'s block sparse attention comes into the picture.
 
-### Understanding with Graphs
+### Understanding Big Bird's attention with Graphs
 
-Let's try to understand the need of `global`, `sliding` & `random` attention using the graphs.
+Let's try to understand the need of `global`, `sliding` & `random` attention using graphs.
 
 <img src="assets/global.png" width=256 height=200> </img>
 <img src="assets/sliding.png" width=256 height=200> </img>
 <img src="assets/random.png" width=256 height=200> </img> <br>
-*Above figure shows `global`, `sliding` & `random` connections respectively in graph, where each node corresponds to token and each dotted-line represents attention score. If no connection is made between 2 tokens, then attention score is assumed to 0.*
 
-BigBird block sparse attention is simply combination of these 3 figures. While in normal attention, all 81 connections (note: total 9 nodes are present) would have been present in above figure. You can simply think of normal attention as all the tokens being global.
+*The above figure shows `global` (left), `sliding` (middle) & `random` (right) connections respectively as a graph. Each node corresponds to a token and each dotted-line represents attention score. If no connection is made between 2 tokens, then attention score is assumed to 0.*
+
+BigBird block sparse attention is simply combination of these 3 figures. While in normal attention, all 36 connections (note: total 9 nodes are present) would have been present in above figure. You can simply think of normal attention as all the tokens attending globally.
```

Comment:
Also remember why you show these graphs - IMO it's nice to show the graphs because you want to explain how full attention <=> fully connected graph <=> each tokens has **direct** access to each other node. => sliding attention only to neighbouring nodes & random attention then helps significantly to shorten the distance from one node to the other. So the graphs should really be used to show how random attention helps to shorten the way it takes for one token to reach another token. I think you should add a couple more sentences here